pkg/addon/controllers/addontemplate/controller.go

```diff
@@ -88,27 +97,98 @@ func NewAddonTemplateController(
 		// easy to mock in unit tests
 		c.runControllerFunc = c.runController
 	}
-	return factory.New().WithInformersQueueKeysFunc(
-		queue.QueueKeyByMetaNamespaceName,
-		addonInformers.Addon().V1alpha1().ClusterManagementAddOns().Informer()).
+
+	controller := factory.New().
+		WithSyncContext(syncCtx).
+		WithInformersQueueKeysFunc(
+			queue.QueueKeyByMetaNamespaceName,
+			addonInformers.Addon().V1alpha1().ClusterManagementAddOns().Informer()).
 		WithBareInformers(
+			addonInformers.Addon().V1alpha1().ManagedClusterAddOns().Informer(),
 			// do not need to queue, just make sure the controller reconciles after the addonTemplate cache is synced
 			// otherwise, there will be "xx-addon-template" not found" errors in the log as the controller uses the
 			// addonTemplate lister to get the template object
 			addonInformers.Addon().V1alpha1().AddOnTemplates().Informer()).
 		WithSync(c.sync).
 		ToController("addon-template-controller", recorder)
+
+	// Add custom event handler for ManagedClusterAddon to filter configReference changes
+	_, err := addonInformers.Addon().V1alpha1().ManagedClusterAddOns().Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			// Only handle configReference updates for AddOnTemplates
+			oldMCA, ok := oldObj.(*addonv1alpha1.ManagedClusterAddOn)
+			if !ok {
+				return
+			}
+			newMCA, ok := newObj.(*addonv1alpha1.ManagedClusterAddOn)
+			if !ok {
+				return
+			}
+
+			// Extract AddOnTemplate configReferences from both old and new
+			oldTemplateRefs := extractAddOnTemplateConfigRefs(oldMCA)
+			newTemplateRefs := extractAddOnTemplateConfigRefs(newMCA)
+
+			// Only process if AddOnTemplate configReferences changed
+			if !reflect.DeepEqual(oldTemplateRefs, newTemplateRefs) {
+				// Queue the addon name to trigger reconciliation
+				c.queue.Add(newMCA.Name)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			mca, ok := obj.(*addonv1alpha1.ManagedClusterAddOn)
```

Comment:
same here, we only need to care template based addon.

Reply:
Fixed.